linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/HeaderPathIdentifier.scala

```diff
@@ -0,0 +1,108 @@
+package io.buoyant.linkerd
+package protocol
+package h2
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.{Dtab, Path, Stack}
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.buoyant.h2.{Headers => H2Headers, LinkerdHeaders, Request}
+import com.twitter.util.Future
+import io.buoyant.router.H2
+import io.buoyant.router.RoutingFactory._
+
+class HeaderPathIdentifier(
+  header: String,
+  segments: Option[Int],
+  pfx: Path,
+  baseDtab: () => Dtab
+) extends Identifier[Request] {
+  import HeaderPathIdentifier._
+
+  private[this] val identifyPath: Request => Future[RequestIdentification[Request]] =
+    segments match {
+      case Some(segments) if segments < 1 =>
+        throw new IllegalArgumentException("segments must be >= 1")
+
+      case Some(segments) =>
+        lazy val tooShort = {
```

Comment:
oh.  variable shadowing rears its ugly head :)

segments doesn't depend on the request at all though, so I feel like this could be rejiggered.